components/datepicker/DatePickerInput.js

```diff
@@ -0,0 +1,173 @@
+import React, { PureComponent } from 'react';
+import PropTypes from 'prop-types';
+import Box from '../box';
+import DayPickerInput from 'react-day-picker/DayPickerInput';
+import Icon from '../icon';
+import InputMetaPropTypes from '../input/InputMetaPropTypes';
+import NavigationBar from './NavigationBar';
+import WeekDay from './WeekDay';
+import { convertModifiersToClassnames } from './utils';
+import { IconCalendarSmallOutline, IconWarningBadgedSmallFilled } from '@teamleader/ui-icons';
+import { TextSmall } from '../typography';
+import cx from 'classnames';
+import omit from 'lodash.omit';
+import theme from './theme.css';
+
+class DatePickerInput extends PureComponent {
+  state = {
+    inputHasFocus: false,
+    selectedDate: null,
+  };
+
+  static getDerivedStateFromProps(props, state) {
+    if (props.selectedDate !== undefined && props.selectedDate !== state.selectedDate) {
+      return {
+        selectedDate: props.selectedDate,
+      };
+    }
+
+    return null;
+  }
+
+  handleBlur = () => {
+    this.setState({ inputHasFocus: false });
+  };
+
+  handleFocus = () => {
+    this.setState({ inputHasFocus: true });
+  };
+
+  handleInputDateChange = date => {
+    this.setState(
+      {
+        selectedDate: date,
+      },
+      () => this.props.onChange(date),
+    );
+  };
+
+  hasError = () => {
+    const { meta } = this.props;
+    return Boolean(meta && meta.error && meta.touched);
+  };
+
+  renderDayPickerInput = () => {
+    const { className, dayPickerProps, disabled, modifiers, readOnly, size, ...others } = this.props;
+    const { selectedDate } = this.state;
+
+    const dayPickerClassNames = cx(theme['date-picker'], theme[`is-${size}`], className);
+
+    const rest = omit(others, ['helpText', 'meta', 'onBlur', 'onChange', 'onFocus']);
+
+    return (
+      <DayPickerInput
+        classNames={theme}
+        dayPickerProps={{
+          className: dayPickerClassNames,
+          classNames: theme,
+          modifiers: convertModifiersToClassnames(modifiers, theme),
+          onDayClick: this.handleInputDateChange,
+          selectedDays: selectedDate,
```

Comment:
PP: Little different naming, `selectedDays` sounds like multiple days, `selectedDate` sounds like 1 day

Reply:
`selectedDays` is the prop I have to pass to the `DayPickerInput`, which is a [third-party component](http://react-day-picker.js.org/api/DayPickerInput). So I cannot change this :)